packages/typespec-ts/src/utils/modelUtils.ts

```diff
@@ -83,7 +83,10 @@ export function isByteOrByteUnion(dpgContext: SdkContext, type: Type) {
 }
 
 function isByteType(schema: any) {
-  return schema.type === "string" && schema.format === "byte";
+  return (
+    schema.type === "string" &&
+    (schema.format === "byte" || schema.format === "binary")
```

Comment:
If it's self defined scalar, we will get the format of binary as special handled.